CMakeLists.txt

```diff
@@ -1,100 +1,171 @@
-catkin_add_pybind11_module(py_two_joint)
-catkin_add_pybind11_module(py_real_finger)
-catkin_add_pybind11_module(py_trifinger
-    trifinger_platform_frontend trifinger_platform_log)
+add_pybind11_module(py_one_joint srcpy/py_one_joint.cpp
+    LINK_LIBRARIES
+        Eigen3::Eigen
+        blmc_robots::blmc_robots
+)
+add_pybind11_module(py_two_joint srcpy/py_two_joint.cpp
+    LINK_LIBRARIES
+        Eigen3::Eigen
+        blmc_robots::blmc_robots
+)
+add_pybind11_module(py_real_finger srcpy/py_real_finger.cpp
+    LINK_LIBRARIES
+        Eigen3::Eigen
+        blmc_robots::blmc_robots
+)
+add_pybind11_module(py_trifinger srcpy/py_trifinger.cpp
+    LINK_LIBRARIES
+        blmc_robots::blmc_robots
+        trifinger_platform_frontend
+        trifinger_platform_log
+)
 
 
 # Demo executable
 add_executable(demo_trifinger_platform src/demo_platform_frontend.cpp)
+target_include_directories(demo_trifinger_platform PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
 target_link_libraries(demo_trifinger_platform
     trifinger_platform_frontend
 )
 
 
+# Installation
+ament_export_interfaces(export_${PROJECT_NAME} HAS_LIBRARY_TARGET)
+
+install(DIRECTORY include/${PROJECT_NAME}/
+        DESTINATION include/${PROJECT_NAME})
+
+install(DIRECTORY config
+        DESTINATION share/${PROJECT_NAME})
+
+install(
+    TARGETS
+        trifinger_platform_frontend
+        trifinger_platform_log
+        demo_trifinger_platform
+    EXPORT export_${PROJECT_NAME}
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    RUNTIME DESTINATION lib/${PROJECT_NAME}
+)
+
+install(
+    PROGRAMS
+        demos/demo_data_logging.py
+        demos/demo_fake_finger.py
+        demos/demo_finger_forward_kinematics.py
+        demos/demo_real_finger.py
+        demos/demo_sparse_position_control.py
+        demos/demo_trifingeredu.py
+        demos/demo_trifinger_platform_log.py
+        demos/demo_trifinger_platform.py
+        demos/demo_trifingerpro.py
+        demos/demo_trifinger.py
+
+        scripts/calibrate_home_offset.py
+        scripts/check_finger_position_control_error.py
+        scripts/construct_object_reset_trajectory.py
+        scripts/demonstrate_trajectory.py
+        scripts/evaluate_log.py
+        scripts/fingeredu_endurance_test.py
+        scripts/joint_friction_calibration.py
+        scripts/plot_logged_data.py
+        scripts/position_control_on_off.py
+        scripts/print_position.py
+        scripts/record_and_play_trajectories.py
+        scripts/replay_trajectory.py
+        scripts/robot_log_dat2csv.py
+        scripts/run_onejoint_tests.py
+        scripts/single_finger_test.py
+        scripts/trifinger_backend.py
+        scripts/trifingerpro_offset_calibration.py
+        scripts/trifingerpro_post_submission.py
+        scripts/trifingerpro_print_push_sensor.py
+        scripts/trifingerpro_random.py
+
+    DESTINATION lib/${PROJECT_NAME}
```

Comment:
I would do a
```
install(
  FILES ${list_files}
  DESTINATION lib/${PROJECT_NAME}
  PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE)
```
and `chmod -x -R demos scripts`

Reply:
I removed the executable flag from the scripts.  There is no need to change the cmake file as `install(PROGRAMS` already makes them executable by default (permissions of the installed scripts are `-rwxr-xr-x`).